docs/design/batching.md

```diff
@@ -0,0 +1,56 @@
+# Batching VolumeSnapshotMover Backups and Restores
+
+## Summary
+The VolumeSnapshotMover controller currently runs on each `volumeSnapshotBackup` 
+and `volumeSnapshotRestore` until completed. The problem with this is that as 
+the number of VolSync processes increases, the time for backup/restore to complete
+can become slow, as diminishing returns increases.
+
+
+## Performance Testing Results
+[Performance testing](https://docs.google.com/document/d/1kPKo46_McEkj0Fu9hdkTxiVDnTh00forFQnscTgiEgg/edit?usp=sharing) 
+found that there is an optimal batch number of 12 for backup. For restore, 
+the results appear to be linear. 
+
+
+## Motivation
+Improve the performance of VolumeSnapshotMover backup and restore by allowing 
+users to specify a batch number of `volumeSnapshotBackups` and 
+`volumeSnapshotRestores` that can be simultaneously `inProgress` at once.
+
+
+## Implementation details
+A configurable value, `dpa.spec.dataMover.batchNumber`, can be used to specify 
```

Comment:
As discussed earlier today, we should split this into two parameters. Perhaps `dpa.spec.dataMover.concurrentBackupVolumes` and `dpa.spec.dataMover.concurrentRestoreVolumes`.

Reply:
We should also note that a value of zero indicates no restriction on how many operations we will allow.

Reply:
We should also have a default number defined in the code which will be used if the user doesn't make any preference. Value TBA -- it could be something like 10 or 12, or we could decide that the default is 0 (i.e. unlimited), but if we have consts defined for `DefaultConcurrentBackupVolumes` and `DefaultConcurrentRestoreVolumes`, then it will be easy to change default behavior later by just redefining the const (to 10 or 12 or whatever for a specific limit, or 0 for unlimited).